### PR TITLE
chore: add different python version to ci

### PR DESCRIPTION
**Pull request by samsja**
Signed-off-by: Sami Jaghouar <sami.jaghouar@hotmail.fr>

# What this pr do

* bump to python 3.8 for testing
* check import and install on different python version

### CODE REVIEW

File: .github/workflows/cd.yml [hunk 1]
```diff
@@ -32,7 +32,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version: [3.7]
+        python-version: [3.8]
```
Comment (JohannesMessner) [inline .github/workflows/cd.yml:35]: Shouldn't this include all versions >= 3.8? Or only for CI?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -124,12 +124,15 @@ jobs:
 
   import-test:
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-env: [3.8, 3.9, '3.10', '3.11']
```
Comment (JohannesMessner) [inline .github/workflows/ci.yml:129]: Some have `''`, other don't, does it make a difference?

**Review by JohannesMessner — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 2 above)
Comment (alaeddine-13) [inline .github/workflows/ci.yml:129]: why some are strings and some are not

Comment (samsja) [reply .github/workflows/ci.yml:129]: fixed

Comment (samsja) [reply .github/workflows/ci.yml:129]: fixed

[the remark this answers is not in the document]
Comment (samsja) [reply .github/workflows/ci.yml:129]: well we don't support it so no. If this breaks because of numpy we cannot do anything

**Review by alaeddine-13 — APPROVED** [review]

**Pull request merged**